### ISSUE TITLE
[BUG]CXMM DDEX order size limit & exchange rate converter error

### ISSUE THREAD

**Issue by moraliang** [CoinAlpha/hummingbot#386]
**Describe the bug**
// A clear and concise description of what the bug is.
hummingbot.market.ddex.ddex_market - ERROR - Order size is less than minimum order size allowed
TypeError: adjust_token_rate() got an unexpected keyword argument 'amount'

**Steps To Reproduce**
// A concise description of the **steps to reproduce the buggy behavior**:

```
########################################################
###   Cross exchange market making strategy config   ###
########################################################

# The following configuations are only required for the
# cross exchange market making strategy

# Exchange and token parameters
maker_market: ddex
taker_market: idex
maker_market_symbol: WETH-DAI
taker_market_symbol: DAI_ETH

# Minimum profitability target required to place an order
# Expressed in decimals: 0.01 = 1% target profit
min_profitability: -0.001

# Maximum order size in terms of quote currency
trade_size_override: 15.00

# Maximum aggregate amount of orders in quote currency
# that are allowed at a better price than Hummingbot's
# order before Hummingbot adjusts its order and pricing
top_depth_tolerance:

- - ^.+(USDT|USDC|USDS|DAI|PAX|TUSD)$
  - 1000
- - ^.+ETH$
  - 10
- - ^.+BTC$
  - 0.5
# Have Hummingbot actively adjust/cancel orders if necessary.
# If set to true, outstanding orders are adjusted if
# profitability falls below min_profitability.
# If set to false, outstanding orders are adjusted if
# profitability falls below cancel_order_threshold.
active_order_canceling: true

# If active_order_canceling = false, this is the profitability/
# loss threshold at which to cancel the order.
# Expressed in decimals: 0.01 = 1% target profit
cancel_order_threshold: -0.5

# An amount in seconds, which is the minimum duration for any
# placed limit orders. Default value = 130 seconds.
limit_order_min_expiration: 130.0
```

**Screenshots**
// If applicable, add screenshots to help explain your problem.
![Screen Shot 2019-06-12 at 5 19 31 PM](https://user-images.githubusercontent.com/14824365/59394894-4b255900-8d36-11e9-9b24-cdc28c7edacc.png)

**Release version**
// Include your bot's version number (Can be found at the bottom left corner of your CLI).
-0.9.0 dev

**Attachments**
// Please attach your config file and log file to help us debug. It would be difficult for us to help you without those! 
[hummingbot_logs_2019-06-12-17-11-03.log](https://github.com/CoinAlpha/hummingbot/files/3283603/hummingbot_logs_2019-06-12-17-11-03.log)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**
**Optional: your ETH address:** (If you want to participate in the bug bounty)

**Comment by mifeng** [comment]
@moraliang Did you deposit tokens into IDEX? This error typically appears when you have no inventory in the `taker_market`. Since the XEMM strategy caps the trade size at the total amount that is hedgeable in the `taker_market`, if there is no inventory there, the trade size will become 0.

Note that unlike other DEXs, you have to deposit tokens into IDEX before trading there.

**Comment by veecos2** [comment]
I just realized that in terms of quote currency you need to specify for 0.15 ETH which is definitely more than 15 DAI. Can you retest this ?

**Comment by nkona** [comment]
Although the bug is corrected for ddex in the latest dev version, I encountered the same "TypeError: adjust_token_rate() got an unexpected keyword argument 'amount'" specifically when using radar as a taker exchange for xemm. This occurred regardless of the maker exchange, and did not occur when radar was the maker exchange or under an arb strategy. I have checked that balances are sufficient, and it appears that other configuration settings do not influence this bug's appearance.
